src/Compilers/VisualBasic/Portable/Scanner/ScannerBuffer.vb

```diff
@@ -94,7 +94,29 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             _curPage = p
             Return p
         End Function
-
+        ''' <summary>
+        ''' Peep at the next index position, if it is reachable pass the char back
+        ''' </summary>
+        ''' <param name="ch">(Passback) the character at that location if possible.
+        ''' If not the contents are left unaffected.
+        ''' NOTE: This is different behaviour to the Try methods that always return a value.
+        ''' </param>
+        ''' <returns>Returns True if it possible to peek at, otherwise returns False.</returns>
+        Friend Function Peep(ByRef ch As Char) As Boolean
+            ' CanGet()
+            If _lineBufferOffset >= _bufferLen Then
+                Return False
+            End If
+            ' Peek()
+            Dim page = _curPage
+            Dim position = _lineBufferOffset
+            ch = page._arr(position And s_PAGE_MASK)
+            If page._pageStart <> (position And s_NOT_PAGE_MASK) Then
+                page = GetPage(position)
+                ch = page._arr(position And s_PAGE_MASK)
+            End If
```

Comment:
@AlekseyTs Perf reasons
Consider in for the cases where `CanGet(  ) AndAlso Peek( )` (a likely one)
We can utilise this and make it a bit more performant;- 
- since index is only need once
- remove the need for the `AndAlso` boolean operator. 
Then in the following two cases..
Case 1: " `CanGet` " = False, is the slightly better as it still one function call, not the two of calling `CanGet` internally. (Effectively it has has been in-lined without the need of but a attribute on `CanGet`. )
 
Case 2: " `CanGet` " = True,
-  is little bit more efficient ( no additional function call )
-  plus we can utilise the fact it is after the canget part and remove some checks.